sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -217,8 +221,32 @@ struct _pi_context : _pi_object {
     // we don't need a map with device as key.
     HostMemAllocContext = new USMAllocContext(
         std::unique_ptr<SystemMemory>(new USMHostMemoryAlloc(this)));
+
+    // If we have a single device/sub-device, construction is done.
+    if (NumDevices < 2)
+      return;
+
+    // Check if we have context with subdevices of the same device (context may
+    // include root device itself as well)
+    RootDevice = Devices[0]->RootDevice ? Devices[0]->RootDevice : Devices[0];
+    for (auto &Device : Devices) {
+      if ((!Device->RootDevice && Device != RootDevice) ||
+          (Device->RootDevice && Device->RootDevice != RootDevice)) {
+        RootDevice = nullptr;
+        break;
+      }
+    }
   }
 
+  // Currently support of multi-device contexts is limited in the Level Zero
+  // plugin. For example, device allocations are supported only for contexts
+  // with a single device or when context consist of sub-devices of the same
+  // device. This method picks a "default" device which is used to allocate
+  // device memory and create immediate command lists to initialize allocations.
+  // It is going to be removed when multi-device contexts will be fully
+  // supported.
+  pi_device defaultDevice() { return RootDevice ? RootDevice : Devices[0]; }
```

Comment:
The comment text is not very clear. It sounds that we don't support context with multiple root-devices of the same driver, but I thought we do. Also the `defaultDevice` name is not speaking to how this function is to be used. It would return Devices[0] for context's with multiple-root devices, but why is this correct?

Reply:
We still don't fully support support contexts with multiple root-devices. We support multi-device contexts for buffers by means of host allocations with this commit. But, for example, images are still not supported, creating a queue from native handle is not supported, so there are cases when we still just use the first device from the context.
Have only sub-devices in the context is a special case which is supported.
This is just utility function, used more clear name.

Reply:
How are the unsupported scenarios reported to the user? I hope we don't just do something that we know wouldn't work.

Reply:
Unfortunately, unsupported cases weren't reported to the user even before this commit.